_blogposts/2023-02-02-release-10-1.mdx

```diff
@@ -0,0 +1,302 @@
+- **Allow mixing JSX configurations on the project and module level.** Gradually mix and match JSX transformations and modes without migrating any old code!
+- **Pass `prop` types** to `@react.component`. You can now fine tune `@react.component` with your specific prop type needs. Very useful for libraries and frameworks to define component interfaces.
+- **Revisited props spread operator.** This will allow users to spread records in JSX without sacrificing their sanity. Note that this implementation has harder constraints than its JS counterpart. (requires `rescript/react@0.11` or higher)
+
+Code tells more than words, so here's a non-exhaustive code example to highlight the different JSX features. Make sure to also check out the JS output and play around with the code in our newest playground!
+
+<CodeTab labels={["ReScript", "JS Output"]}>
+
+```res
+// Set the jsx configuration per module
+@@jsxConfig({version: 4, mode: "automatic"})
+
+module AutomaticModeExample = {
+  // "automatic" mode will compile jsx to the React 18 compatible
+  // jsx-runtime calls
+  @@jsxConfig({version: 4, mode: "automatic"})
+
+  @react.component
+  let make = (~name) => {
+    <div> {React.string(`Hello ${name}`)} </div>
+  }
+}
+
+module ClassicModeExample = {
+  // "classic" mode will compile jsx to React.createElement calls
+  @@jsxConfig({version: 4, mode: "classic"})
+
+  @react.component
+  let make = (~name) => {
+    <div> {React.string(`Hello ${name}`)} </div>
+  }
+}
+
+module NoDecoratorExample = {
+  // No need for `makeProps` anymore
+  type props = {name: string}
+
+  let make = (props: props) => {
+    <div> {React.string(`Hello ${props.name}`)} </div>
+  }
+}
+
+module ReactInterfaceExample: {
+  @react.component
+  let make: (~name: string, ~age: int=?) => React.element
+} = {
+  @react.component
+  let make = (~name, ~age=0) => {
+    <div>
+      {React.string(
+        `Hello ${name}, you are ${Belt.Int.toString(age)} years old.`,
+      )}
+    </div>
+  }
+}
+
+module PropTypeInjectionExample = {
+  // Let's assume we have a prop type that we wanna enforce
+  // as our labeled arguments
+  type someoneElsesProps = {isHuman: bool}
+
+  // Here we tell the `react.component` decorator what props to infer.
+  // Useful for e.g. NextJS usage, or to create components that should
+  // comply to certain library component interfaces
+  @react.component(: someoneElsesProps)
+  let make = (~isHuman) => {
+    let msg = switch isHuman {
+    | true => "hello human"
+    | false => "hello fellow computer"
+    }
+    <div> {React.string(msg)} </div>
+  }
+}
+
+module PropSpreadExample = {
+  // Note: This will require @rescript/react 0.11 or later
+  @@jsxConfig({version: 4, mode: "automatic"})
+
+  @react.component
+  let make = () => {
+    let props = {NoDecoratorExample.name: "World"}
+
+    <NoDecoratorExample {...props} />
+  }
+}
+
```

Comment:
Add another example for better type inference on nominal types:
https://rescript-lang.org/try?version=v10.1.2&code=ALBWGcA8GEHsDsBmBLA5gCgN4AIBuBTAJ3GQQC5sBmAGmwFtYATfCgIgEMBXAF1jve7IAxq2wBfAJQAoKQ0acANvmwBVcEWwBebJinZs3AJ4AHZeG4DO4LdgDy8BcnjKAPncSJHzvdh-BC+OxC3AB0QnzGCPjw3D5K3PTsANbK2ugAfuaW4BJaAHw6PvrgAO7I3EIAFthZ3FaF+o3YbvZeqQUASoHBIfCKCkWNLR5t+dhdQaF9CgNNYj7z81Lx2AD6NgA8ahq1VpqtTsoA9HlSQA

Reply:
no need anymore, the improvement has been mentioned above already.